tests/utils/test_bitwise_casts.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+
+from xdsl.utils.bitwise_casts import convert_f32_to_u32, convert_u32_to_f32
+
+
+# http://bartaz.github.io/ieee754-visualization/
+@pytest.mark.parametrize(
+    "i, f",
+    [
+        (0b00000000000000000000000000000000, 0.0),
+        (0b10000000000000000000000000000000, -0.0),
+        (0b00111111100000000000000000000000, 1.0),
+        (0b01000000000000000000000000000000, 2.0),
+    ],
+)
+def test_float_bitwise_casts(i: int, f: float):
+    # assert i == 0
+    assert convert_f32_to_u32(f) == i
+    # N.B. this is an IEEE float comparison, 0.0 == -0.0, but still a useful test
```

Comment:
Again what is this comment about?

Reply:
Just a reminder to people reading the code that it's not a perfect comparison

Reply:
Clarified it a little

Reply:
So we want bit-wise float comparison? Or something else?

Reply:
It would be nice but there is no such thing. It's explicitly not a supported operation, and converting both operands to int and comparing ints is the recommended way to do it, sort of what this test is testing in the first place.

Reply:
See requested changes for bit-wise float compare.

Reply:
Good idea